Automattic/jetpack
# Contact Form block updates

Fixes #15286.

Update the form block to support more than a standard contact form using the variations API. Clean up and improve a number of visual and block settings.

## Full list of changes:

✔Supports the variations API to provide a Contact, RSVP, Registration, Appointment, and Feedback form template that also appear as separate blocks.
✔️No more settings placeholder before seeing the form, the block now uses your admin account email by default.
✔️Form settings can now be accessed via the block toolbar (as well as the sidebar).
✔️The email subject changes its value depending on the type of form variation you are using.
✔️The required setting for a field is no longer on the canvas, but in the block toolbar and sidebar.
✔️The form field label no longer has a fixed white background and does not jump when selected.
✔️Removed the "default value" setting for fields because this simply did not function.
✔️Moved the CSS ID to the Advanced panel in the sidebar for fields
✔️Added a bunch of settings help text to guide users through settings.
✔️Improved the size and spacing of form field elements in the front end.
✔️Made form field elements non-editable in the editor so that you can click on them and select the actual block instead.
✔️Added support for using the new Jetpack button block as an innerblock in the form.

## Screenshots:

Before                                   |  After
:-------------------------:|:-------------------------:
<img width="660" alt="Screen Shot 2020-04-22 at 2 45 59 PM" src="https://user-images.githubusercontent.com/1464705/80037422-66113f80-84a8-11ea-8342-058579c45cb6.png"> | <img width="680" alt="Screen Shot 2020-04-22 at 2 44 08 PM" src="https://user-images.githubusercontent.com/1464705/80037433-6ad5f380-84a8-11ea-98eb-e66cc2755899.png">
<img width="664" alt="Screen Shot 2020-04-22 at 2 46 10 PM" src="https://user-images.githubusercontent.com/1464705/80037466-79240f80-84a8-11ea-948a-67c60e71c657.png"> | <img width="656" alt="Screen Shot 2020-04-22 at 2 45 17 PM" src="https://user-images.githubusercontent.com/1464705/80037453-745f5b80-84a8-11ea-9958-2e415d6fdc06.png"> 

Fixes #15286 

## Testing instructions:
* Insert a regular "Form" block and select a variation. Confirm this works as expected.
* Check the form on the front end and confirm the display matches and the form submits correctly.
* Check the Form Settings in the block toolbar and make sure the default email and subject lines are filled and correct.
* Modify the required setting for form fields and confirm this works as expected.
* Add a form variation directly from the block inserter (for example choose RSVP form) and confirm this inserts and works as expected.
* Using the master branch, insert a contact form using the previous version of the block and confirm it still works in the editor and front end when switching to this branch.

#### Proposed changelog entry for your changes:

* Added new blocks for RSVP, Registration, Appointment, and Feedback forms for your website.
